Assets/_PartyParkade/Script/Debug/Debug Logic/PowerupDebuggerLogic.cs

```
@@ -0,0 +1,145 @@
+﻿//-----------------------------------------------------------------------
+// <copyright file="PowerupDebuggerLogic.cs" company="PERIGON GAMES">
+//     Copyright (c) PERIGON GAMES. 2019 All Rights Reserved
+// </copyright>
+//-----------------------------------------------------------------------
+using System;
+using System.Collections.Generic;
+using _PartyParkade.Script.Game;
+using _PartyParkade.Script.Data;
+using UnityEngine;
+
+public class PowerupDebuggerLogic : DebugLogic
+{
+    #region Rects
+    private Rect m_titleRect;
+    private Rect m_selectionRect;
+    private Rect m_removeAllRect;
+    #endregion
+
+    private List<PlayerController> m_players;
+    private float[] m_sliderValues;
+    private int m_numberOfPowerups;
+
+    public PowerupDebuggerLogic(Rect masterRect, bool isDebugBuild) : base(masterRect, isDebugBuild) { }
+
+    public override void InitializeData()
+    {
+        m_players = PlayerManager.Players != null ? PlayerManager.Players : null;
+        m_sliderValues = new float[m_players.Count];
```

Why would PlayerManager.Players be null? It's a list isn't it?

---

A list can be null!!

---

Yes, but if it's null then we have a bigger problem don't you think? Like that should raise an error in the PlayerManager if its list of Players is null.

---

And then we don't have to be checking if its null all over our code because we can be sure that whoever designed the PlayerManager designed it in such a way that errors with its own variables are handled in the PlayerManager class itself.

---

And if we are doing null checks anywhere in our code like this then we should probably be logging something at the very least to say that its null.

---

I think that even though you never set it to null, you should still do null checks. 

It's part of defensive programming where, you never expected it to be null, but just incase it is null, it'll still run.

---

We should be logging in cases where it is null then. Like add an `else` statement to do a Debug.Log or Error.Log or something

---

yup, an else to Debu.log sounds like the way to go!!